allspice/utils/list_components.py

```diff
@@ -452,6 +476,40 @@ def _extract_components(
     return components
 
 
+def _combine_multi_part_components_for_altium(
+    components: list[dict[str, str]],
+) -> list[dict[str, str]]:
+    """
+    Combine multi-part Altium components into a single component.
+
+    Altium multi-part components can be distinguished by the `_part_count` and
+    `_current_part_id` attributes being present, which respectively store the
+    total number of parts and the current part number. If that is the case, the
+    `_logical_designator` attribute ties together the different parts of the
+    component.
+    """
+
+    combined_components = []
+    multi_part_components_by_designator = {}
+
+    for component in components:
+        if "_part_count" in component and "_current_part_id" in component:
+            designator = component["_logical_designator"]
+            multi_part_components_by_designator.setdefault(designator, []).append(component)
+        else:
+            combined_components.append(component)
+
+    for designator, multi_part_components in multi_part_components_by_designator.items():
+        combined_component = multi_part_components[0].copy()
```

Comment:
There is an interesting side effect of this statement. We're taking the attributes of the "first" part as the attributes of the combined component. But there can be cases where the attributes differ. In Archimajor, there are some places where the first part has no attributes but some other part does. I don't think there is a reasonable and straightforward solution here for that case, which I think should be considered an issue on the board.